src/System/IO/Resource/Linear/Internal.hs

```diff
@@ -98,101 +113,87 @@ instance Control.Monad RIO where
 
 -- files
 
--- Remark: Handle needs to be private otherwise `Data.Coerce.coerce` could wreak
--- Havoc on the abstraction. But we could provide a smart constructor/view to
--- unsafely convert to file handles in order for the Handle API to be
--- extensible.
-
-newtype Handle = Handle (UnsafeResource System.Handle)
+type Handle = Resource System.Handle
 
 -- | See @System.IO.'System.IO.openFile'@
 openFile :: FilePath -> System.IOMode -> RIO Handle
-openFile path mode = Control.do
-  h <-
-    unsafeAcquire
-      (Linear.fromSystemIOU $ System.openFile path mode)
-      (\h -> Linear.fromSystemIO $ System.hClose h)
-  Control.return $ Handle h
+openFile path mode =
+  unsafeAcquire
+    (Linear.fromSystemIOU $ System.openFile path mode)
+    (\h -> Linear.fromSystemIO $ System.hClose h)
 
 -- | See @System.IO.'System.IO.openBinaryFile'@
 --
 -- @since 0.3.0
 openBinaryFile :: FilePath -> System.IOMode -> RIO Handle
-openBinaryFile path mode = Control.do
-  h <-
-    unsafeAcquire
-      (Linear.fromSystemIOU $ System.openFile path mode)
-      (\h -> Linear.fromSystemIO $ System.hClose h)
-  Control.pure $ Handle h
+openBinaryFile path mode =
+  unsafeAcquire
+    (Linear.fromSystemIOU $ System.openFile path mode)
+    (\h -> Linear.fromSystemIO $ System.hClose h)
 
+-- | Specialised alias for 'release'
 hClose :: Handle %1 -> RIO ()
-hClose (Handle h) = unsafeRelease h
+hClose = release
```

Comment:
Deprecate this? Not with a view to ever removing it, but programmers put `hClose` in their programs, then get told to use `release`, and then learn not to provide aliases in their own libraries?

Reply:
I don't know that I want to deprecate this. I would if I thought using `hClose` is less good than using `release`. But I'm not sure that it is, just it's ok that we have `return` despite it being a specialisation of `pure`.